### PR TITLE
CP-35838: Add health check precheck into RPU wizard

### PR DESCRIPTION
**Pull request by jijiang**
Signed-off-by: Ji Jiang <ji.jiang@citrix.com>

### CODE REVIEW

**Review by MihaelaStoica — CHANGES_REQUESTED** [review]

File: XenModel/Messages.resx [hunk 1]
```diff
@@ -2651,6 +2651,9 @@ Do you want to assign it to the schedule '{2}' instead?</value>
   <data name="CHECKING_HA_STATUS" xml:space="preserve">
     <value>Checking HA and WLB status</value>
   </data>
+  <data name="CHECKING_HEALTH_CHECK_SERVICE" xml:space="preserve">
+    <value>Checking pool health check service status</value>
```
Comment (kc284) [inline XenModel/Messages.resx:2655]: I think `Checking pool enrollment to the Health Check service` is more accurate.

File: XenModel/Messages.resx [hunk 2]
```diff
@@ -10657,6 +10660,12 @@ Please reconnect the host and try again</value>
   <data name="PROBING_HBA_TITLE" xml:space="preserve">
     <value>Probing for LUNs on {0}</value>
   </data>
+  <data name="PROBLEM_HEALTH_CHECK_SERVICE_DESCRIPTION" xml:space="preserve">
```
Comment (kc284) [inline XenModel/Messages.resx:10663]: On the UI "Health Check" is used with capital initial letters as a term, so let's follow this convention here too (and in the messages below). Also, it needs the article.

File: XenAdmin/Diagnostics/Problems/PoolProblem/HealthCheckServiceProblem.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ */
+
+using XenAdmin.Core;
+using XenAdmin.Diagnostics.Checks;
+using XenAPI;
+
+
+namespace XenAdmin.Diagnostics.Problems.PoolProblem
+{
+    class HealthCheckServiceProblem : ProblemWithMoreInfo
```
[the remark this answers is not in the document]
Comment (jijiang) [reply XenAdmin/Diagnostics/Problems/PoolProblem/HealthCheckServiceProblem.cs]: Fixed.

File: XenModel/Messages.resx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jijiang) [reply XenModel/Messages.resx]: Fixed.

Comment (jijiang) [reply XenModel/Messages.resx]: Fixed.

File: XenModel/Messages.resx (hunk 2 above)
Comment (jijiang) [reply XenModel/Messages.resx:10663]: Fixed.

**Comment by kc284** [comment]
I downloaded the commit and had a look around. It works nice, however, I realised we have a usability issue. When the users RPU to an unknown version and see the warning, they have no way to disable health check because we don't show the dilalog and there is no action like the problem case. So I think we need to offer the option on the MoreInfo pop-up in case they need to disable it. Let's try adding a `public virtual AsyncAction LinkAction => null;` in WarningWithMoreInfo.cs, then in HealthCheckServiceWarning override it and also set LinkText, and in WarningWithMoreInfo.cs where we launch that warning dialog offer it as an alternative to the LinkData, i.e.
```
if (!string.IsNullOrEmpty(LinkText) && (!string.IsNullOrEmpty(LinkData) || LinkAction != null))
{
    dlg.LinkText = LinkText;
    if (!string.IsNullOrEmpty(LinkData))
        dlg.LinkData = LinkData;
    else
        dlg.LinkAction = LinkAction;
    dlg.ShowLinkLabel = true;
}
```

**Comment by kc284** [comment]
A simpler alternative may be to override CreateAction() in the HealthCheckServiceWarning.cs and show a WarningDialog with the more info message and two buttons, DisableHealthCheck and Cancel.

File: XenModel/Messages.resx [hunk 4]
```diff
@@ -10657,6 +10660,12 @@ Please reconnect the host and try again</value>
   <data name="PROBING_HBA_TITLE" xml:space="preserve">
     <value>Probing for LUNs on {0}</value>
   </data>
+  <data name="PROBLEM_HEALTH_CHECK_SERVICE_DESCRIPTION" xml:space="preserve">
+    <value>{0}: The Health Check service has been removed in [XenServer] {1}.</value>
```
Comment (kc284) [inline XenModel/Messages.resx:10664]: Could you please also change `[XenServer] {1}` to `{0} {1}` here and below, and format it with the BrandManager variables? We should start migrating from rebranding with sed.

**Review by AaronRobson — APPROVED** [review]

File: XenModel/Messages.resx (hunk 4 above)
Comment (jijiang) [reply XenModel/Messages.resx]: Fixed in second commit.

**Review by kc284 — APPROVED** [review]

**Review by MihaelaStoica — APPROVED** [review]

**Pull request merged**